Wrappers/Python/cil/optimisation/operators/Operator.py

```diff
@@ -188,15 +216,21 @@ def PowerMethod(operator, max_iteration=10, initial=None, tolerance = 1e-5,  ret
         2.0005647295658866
 
         """
+        convergence_check = True
+        
+        if method not in ["auto","direct_only","composed_with_adjoint"]:
+            ValueError(" The flag 'method' should take one of 'auto','direct_only','composed_with_adjoint'")

```

Comment:
Duplicating the list of strings can be avoided. And you need to `raise` the error.

```suggestion
        
        allowed_methods = ["auto","direct_only","composed_with_adjoint"]
        
        if method not in allowed_methods:
            raise ValueError("The argument 'method' can be set to one of {0} got {1}".format(allowed_methods, method))

```

Reply:
Thanks!